internal/datanode/channel_meta.go

```diff
@@ -606,3 +634,132 @@ func (c *ChannelMeta) listNotFlushedSegmentIDs() []UniqueID {
 
 	return segIDs
 }
+
+func (c *ChannelMeta) getChannelCheckpoint(ttPos *internalpb.MsgPosition) *internalpb.MsgPosition {
+	c.segMu.RLock()
+	defer c.segMu.RUnlock()
+	channelCP := &internalpb.MsgPosition{Timestamp: math.MaxUint64}
+	// 1. find the earliest startPos in current buffer and history buffer
+	for _, seg := range c.segments {
```

Comment:
Careful, there might be compacted segments, Please check segment validation before any operation

Reply:
thanks for reminding